### DISCUSSION TITLE
问问大家，不太懂这个webpush推送，iOS和Android都不行捏

### DISCUSSION THREAD

**Discussion by cng10001** [jxxghp/MoviePilot#2311] [问题咨询]
iOS和Android都装了mp的web应用 通知都开启的

后台日志显示是fcm没通 加了代理规则了 不行
![image](https://github.com/jxxghp/MoviePilot/assets/115156545/b2621856-70c2-442e-bc36-829e4fad346b)
![image](https://github.com/jxxghp/MoviePilot/assets/115156545/aea9b9a0-f0dc-4b52-ba62-b3fb62ed5163)
![Screenshot_2024-06-10-20-36-16-295_com google android gms-edit](https://github.com/jxxghp/MoviePilot/assets/115156545/57e296e7-fda6-4309-8a7f-6eb77c35276c)
![Screenshot_2024-06-10-20-39-27-127_com miui securitycenter](https://github.com/jxxghp/MoviePilot/assets/115156545/18b8bfc4-e486-4fd3-9513-17fd690a61e4)
![image](https://github.com/jxxghp/MoviePilot/assets/115156545/7c7aa20c-55b2-441a-8f1c-6954fa9ffa91)

**剩下ios的话不是谷歌的fcm了吧？推送不会被墙的吧，也没收到通知。**

![C9D71FEE-398F-4D16-8BF9-28018F5DC7E7](https://github.com/jxxghp/MoviePilot/assets/115156545/2bb2d1d1-7cb5-4221-990c-dafb39ea2635)

我是群晖nas
不知道还有其他方面要设置的吗 问问大家了

**Comment by developer-wlj** [comment]
安卓端，不止要手机代理，mp后端也要代理。你这很明显mp服务端，请求fcm没成功，加代理即可。还有你这不是非标准的https，不知道在不同浏览器上的service work支不支持

**Comment by cng10001** [comment]
环境变量之前只设置了proxy_host,加了https_proxy代理，iOS端能收到了。
但是安卓还是不行，估计还是国内本身用fcm服务就不太好吧搞不像了- -就这样吧

**Comment by developer-wlj** [comment]
虽然没抓包，测试了下，安卓端接受消息是有浏览器的service work维护的，走的并不是系统fcm，需要浏览器保活才可以，当然也需要挂代理
![Screenshot_2024-06-14-11-43-19-960_com.miui.home.jpg](https://github.com/jxxghp/MoviePilot/assets/55836679/c767c3bc-80bd-4acb-bb1e-dd3ac886d16b)

**Reply by cng10001** [reply]
小米的系统吗，具体需要什么设置呢？

**Reply by cng10001** [reply]
额，我怎么感觉我不开代理反而能收到推送~
